scripts/upgrade.sh

```
@@ -7,27 +7,101 @@ LATEST_HELPER_VERSION=${2:-latest}
+log "Extracting all images from docker-compose configuration..."
+COMPOSE_FILES="-f /data/coolify/source/docker-compose.yml -f /data/coolify/source/docker-compose.prod.yml"
+
+# Check if custom compose file exists
+if [ -f /data/coolify/source/docker-compose.custom.yml ]; then
+    COMPOSE_FILES="$COMPOSE_FILES -f /data/coolify/source/docker-compose.custom.yml"
+    log "Including custom docker-compose.yml in image extraction"
+fi
+
+# Get all unique images from docker compose config
+# LATEST_IMAGE env var is needed for image substitution in compose files
+IMAGES=$(LATEST_IMAGE=${LATEST_IMAGE} docker compose --env-file "$ENV_FILE" $COMPOSE_FILES config --images 2>/dev/null | sort -u)
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)